File: backend/app/models/department.py
"""部門資料模型。"""

import uuid
from datetime import datetime

from sqlalchemy import String
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class Department(Base):
    """部門資料表。"""

    __tablename__ = "Departments"

    GUID: Mapped[str] = mapped_column(
        String, primary_key=True, default=lambda: str(uuid.uuid4())
    )
    DeptCode: Mapped[str] = mapped_column(String, nullable=False)
    DeptName: Mapped[str] = mapped_column(String, nullable=False)
    CreateTime: Mapped[datetime] = mapped_column(default=datetime.utcnow)
    UpdateTime: Mapped[datetime] = mapped_column(
        default=datetime.utcnow, onupdate=datetime.utcnow
    )

    # Relationships
    employees = relationship("Employee", back_populates="department")
    schedules = relationship("Schedule", back_populates="department")
    flex_settings = relationship("FlexSetting", back_populates="department")
    required_configs = relationship("RequiredConfig", back_populates="department")

File: frontend/src/types/department.ts
import { z } from "zod";

export interface Department {
  GUID: string;
  DeptCode: string;
  DeptName: string;
  CreateTime: string;
  UpdateTime: string;
}

export const departmentFormSchema = z.object({
  DeptCode: z.string().min(1, "部門代碼為必填"),
  DeptName: z.string().min(1, "部門名稱為必填"),
});

export type DepartmentFormValues = z.infer<typeof departmentFormSchema>;
